charts/orchestrator/templates/tekton-tasks.yaml

```diff
@@ -0,0 +1,265 @@
+      script: |
+        #!/usr/bin/env sh
+        set -eu
+
+        if [ "${PARAM_VERBOSE}" = "true" ] ; then
+          set -x
+        fi
+
+        if [ "${WORKSPACE_BASIC_AUTH_DIRECTORY_BOUND}" = "true" ] ; then
+          cp "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/.git-credentials" "${PARAM_USER_HOME}/.git-credentials"
+          cp "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/.gitconfig" "${PARAM_USER_HOME}/.gitconfig"
+          chmod 400 "${PARAM_USER_HOME}/.git-credentials"
+          chmod 400 "${PARAM_USER_HOME}/.gitconfig"
+        fi
+
+        if [ "${WORKSPACE_SSH_DIRECTORY_BOUND}" = "true" ] ; then
+          cp -R "${WORKSPACE_SSH_DIRECTORY_PATH}" "${PARAM_USER_HOME}"/.ssh
+          chmod 700 "${PARAM_USER_HOME}"/.ssh
+          chmod -R 400 "${PARAM_USER_HOME}"/.ssh/*
+        fi
+
+        # Setting up the config for the git.
+        git config --global user.email "$(params.GIT_USER_EMAIL)"
+        git config --global user.name "$(params.GIT_USER_NAME)"
+
+        eval '$(params.GIT_SCRIPT)'
+
+        RESULT_SHA="$(git rev-parse HEAD | tr -d '\n')"
+        EXIT_CODE="$?"
+        if [ "$EXIT_CODE" != 0 ]
+        then
+          exit $EXIT_CODE
+        fi
+        # Make sure we don't add a trailing newline to the result!
+        printf "%s" "$RESULT_SHA" > "$(results.commit.path)"
+      # Patch to apply on OpenShift
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 65532
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: flattener
+  namespace: {{ .Values.orchestrator.namespace }}
+spec:
+  workspaces:
+    - name: workflow-source
+  params:
+    - name: workflowId
+      description: The workflow ID from the repository
+      type: string
+    - name: convertToFlat
+      description: Whether conversion to flat layout is needed or it's already flattened
+      type: string
+      default: "true"
+  steps:
+    - name: flatten
+      image: registry.access.redhat.com/ubi9-minimal
+      workingDir: $(workspaces.workflow-source.path)
+      script: |
+        echo "--pwd"
+        pwd
+        echo "--ls"
+        ls
+        echo "--ls /workspace/*"
+        ls /workspace/*
+        ROOT=/workspace/workflow
+        TARGET=flat
+        mkdir -p flat
+        if [ "$(params.convertToFlat)" == "true" ]; then
+          if [ -d "workflow/$(params.workflowId)" ]; then
+            cp -r workflow/$(params.workflowId)/src/main/resources flat/$(params.workflowId)
+          else
+            cp -r workflow/src/main/resources flat/$(params.workflowId)
+          fi
+        else
+          cp -r workflow/$(params.workflowId) flat/$(params.workflowId)
+        fi
+
+        curl -L https://raw.githubusercontent.com/parodos-dev/serverless-workflows/main/pipeline/workflow-builder.Dockerfile -o flat/workflow-builder.Dockerfile
+
+        echo "--ls /workspace"
+        ls /workspace
+        echo "--ls /workspace/*"
+        ls /workspace/*
+        echo "--ls -l flat"
+        ls -l flat
```

Comment:
you surely like to print where you are and what you have in the folders :)

Reply:
You never know where you are when a Task is executing, better to ask 😉

Reply:
If it's to provide feedback, keep them but make the message more explicit :)